src/frontends/pytorch/src/op/to.cpp

```diff
@@ -24,22 +24,44 @@ OutputVector translate_to(NodeContext& context) {
         // -> (Tensor(a))
         dtype_idx = 1;
         memory_format_idx = 4;
+        auto node = context.get_input_from_visible_context(dtype_idx).get_node_shared_ptr();
+        auto fw_node = std::dynamic_pointer_cast<PtFrameworkNode>(node);
+        if (fw_node && fw_node->get_op_type() == "prim::device") {
+            // Cast only to device without changing dtype. Return input node unchanged.
+            return {context.get_input(0)};
+        }
+        if (fw_node && fw_node->get_op_type() == "prim::Constant") {
+            // Device param can be set using constant.
+            if (context.get_input_type(dtype_idx).is<type::Tensor>() == false) {
```

Comment:
Why "== false" instead of just "!context.get_input_type(dtype_idx).is<type::Tensor>()" ?

Reply:
Fixed